File: output/pascal/najwiekszy_wspolny_dzielnik.pas
program NajwiekszyWspolnyDzielnik ;

var
    a : integer ;
    b : integer ;
    temp : integer ;

begin
    writeln ( "Podaj pierwsza liczbe: " ) ;
    readln ( a ) ;
    writeln ( "Podaj druga liczbe: " ) ;
    readln ( b ) ;

    while b <> 0 do
        begin
            temp := b ;
            b := a mod b ;
            a := temp ;
        end ;

    writeln ( "Najwiekszy wspolny dzielnik to: " ) ;
    writeln ( a ) ;
end .

File: output/c/najwiekszy_wspolny_dzielnik.c
#include <stdio.h>

/* Program: NajwiekszyWspolnyDzielnik */
int main(void) {
    int a;
    int b;
    int temp;
    
    printf("Podaj pierwsza liczbe: ");
    scanf("%d", &a);
    printf("Podaj druga liczbe: ");
    scanf("%d", &b);
    while ((b != 0)) {
        temp = b;
        b = (a % b);
        a = temp;
    }
    printf("Najwiekszy wspolny dzielnik to: ");
    printf("%d\n", a);
    return 0;
}
